README.md

```diff
@@ -232,6 +233,26 @@ If it is not in the current working directory along with the indexer command we
 ~$ ./algorand-indexer daemon -c <full-file-location>/indexer.yml
 ```
 
+## Data Directory
```

Comment:
I think it makes sense to guide people towards storing things in the data directory. To that end it might be good to:
1) put this section ahead of the config file section.
2) remove the bullet points about how config files are discovered if none are provided (or mark it as deprecated).